File: scripts/generate_secrets.js
#!/usr/bin/env node
/**
 * Script pour générer des secrets sécurisés pour l'authentification
 */
const crypto = require('crypto');
const fs = require('fs');
const readline = require('readline');

function generateSecret(length = 32) {
  /**
   * Génère un secret sécurisé de longueur spécifiée (en bytes)
   * Retourne une chaîne hexadécimale
   */
  return crypto.randomBytes(length).toString('hex');
}

function main() {
  console.log('='.repeat(50));
  console.log('Génération de secrets sécurisés');
  console.log('='.repeat(50));
  console.log();

  // Générer NEXTAUTH_SECRET (32 bytes = 64 caractères hex)
  const nextauthSecret = generateSecret(32);
  console.log(`NEXTAUTH_SECRET=${nextauthSecret}`);
  console.log();

  // Générer SECRET_KEY (32 bytes = 64 caractères hex)
  const secretKey = generateSecret(32);
  console.log(`SECRET_KEY=${secretKey}`);
  console.log();

  // Générer un secret pour JWT (32 bytes = 64 caractères hex)
  const jwtSecret = generateSecret(32);
  console.log(`JWT_SECRET=${jwtSecret}`);
  console.log();

  console.log('='.repeat(50));
  console.log('Copiez ces valeurs dans vos fichiers .env');
  console.log('='.repeat(50));
  console.log();
  console.log('⚠️  IMPORTANT: Ne partagez JAMAIS ces secrets !');
  console.log('⚠️  Stockez-les de manière sécurisée (variables d\'environnement, gestionnaires de secrets)');
  console.log();

  // Option pour sauvegarder dans un fichier (non versionné)
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  });

  rl.question('Voulez-vous sauvegarder ces secrets dans un fichier .secrets.local ? (o/N): ', (answer) => {
    if (answer.toLowerCase() === 'o') {
      const content = `NEXTAUTH_SECRET=${nextauthSecret}\nSECRET_KEY=${secretKey}\nJWT_SECRET=${jwtSecret}\n`;
      fs.writeFileSync('.secrets.local', content);
      console.log('✅ Secrets sauvegardés dans .secrets.local');
      console.log('⚠️  Assurez-vous que ce fichier est dans .gitignore !');
    }
    rl.close();
  });
}

main();

File: scripts/generate_secrets.py
#!/usr/bin/env python3
"""
Script pour générer des secrets sécurisés pour l'authentification
"""
import secrets
import sys

def generate_secret(length=32):
    """
    Génère un secret sécurisé de longueur spécifiée (en bytes)
    Retourne une chaîne hexadécimale
    """
    return secrets.token_hex(length)

def main():
    print("=" * 50)
    print("Génération de secrets sécurisés")
    print("=" * 50)
    print()
    
    # Générer NEXTAUTH_SECRET (32 bytes = 64 caractères hex)
    nextauth_secret = generate_secret(32)
    print(f"NEXTAUTH_SECRET={nextauth_secret}")
    print()
    
    # Générer SECRET_KEY (32 bytes = 64 caractères hex)
    secret_key = generate_secret(32)
    print(f"SECRET_KEY={secret_key}")
    print()
    
    # Générer un secret pour JWT (32 bytes = 64 caractères hex)
    jwt_secret = generate_secret(32)
    print(f"JWT_SECRET={jwt_secret}")
    print()
    
    print("=" * 50)
    print("Copiez ces valeurs dans vos fichiers .env")
    print("=" * 50)
    print()
    print("⚠️  IMPORTANT: Ne partagez JAMAIS ces secrets !")
    print("⚠️  Stockez-les de manière sécurisée (variables d'environnement, gestionnaires de secrets)")
    print()
    
    # Option pour sauvegarder dans un fichier (non versionné)
    save = input("Voulez-vous sauvegarder ces secrets dans un fichier .secrets.local ? (o/N): ")
    if save.lower() == 'o':
        with open('.secrets.local', 'w') as f:
            f.write(f"NEXTAUTH_SECRET={nextauth_secret}\n")
            f.write(f"SECRET_KEY={secret_key}\n")
            f.write(f"JWT_SECRET={jwt_secret}\n")
        print("✅ Secrets sauvegardés dans .secrets.local")
        print("⚠️  Assurez-vous que ce fichier est dans .gitignore !")

if __name__ == "__main__":
    main()

